src/work-allocation/resolvers/task.resolver.ts

```diff
@@ -1,25 +1,30 @@
 import { Injectable } from '@angular/core';
 import { ActivatedRouteSnapshot, Resolve, Router, RouterStateSnapshot } from '@angular/router';
-import { EMPTY, Observable } from 'rxjs';
+import { combineLatest, EMPTY, Observable } from 'rxjs';
 import { catchError } from 'rxjs/operators';
+import { Caseworker } from '../models/dtos';
 
 import { Task } from '../models/tasks';
-import { WorkAllocationTaskService } from '../services';
+import { CaseworkerDataService, WorkAllocationTaskService } from '../services';
 import { handleFatalErrors, WILDCARD_SERVICE_DOWN } from '../utils';
 
 @Injectable({ providedIn: 'root' })
-export class TaskResolver implements Resolve<Task> {
+export class TaskResolver implements Resolve<any> {
   constructor(
     private readonly service: WorkAllocationTaskService,
-    private readonly router: Router
+    private readonly router: Router,
+    private readonly caseworkerService: CaseworkerDataService
   ) {}
 
-  public resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<Task> {
-    return this.service.getTask(route.paramMap.get('taskId')).pipe(
+  public resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<any> {
+    const task$ =  this.service.getTask(route.paramMap.get('taskId')).pipe(
       catchError(error => {
         handleFatalErrors(error.status, this.router, WILDCARD_SERVICE_DOWN);
         return EMPTY;
       })
     );
+    const caseworker$ = this.caseworkerService.getAll();
+
+    return combineLatest([task$, caseworker$]);
```

Comment:
I think this should be a `forkJoin` to ensure that _both_ have completed before emitting anything.

Reply:
Agree forkJoin is more appropriate here. Will change it

Reply:
Done